<launch>
	<arg name="projector_camera_id" default="0"/>
	<arg name="camera" default="projector_cam"/>

	<arg name="ip"/>
	<arg name="port"/>

	<param name="study_data_dir" value="$(find projector_interface)/study/data"/>

	<!-- config -->
	<include file="$(find projector_interface)/study/launch/setup.launch"/>

	<param name="/screen/border_x" type="double" value="0.35"/>
	<param name="/screen/border_y" type="double" value="0.35"/>

	<!-- webcam (only used for camera_info) -->
	<include file="$(find simple_webcam)/launch/camera_pipeline.launch">
		<arg name="camera_id" value="$(arg projector_camera_id)"/>
		<arg name="info_url" value="file://$(find projector_interface)/study/cfg/camera.yaml"/>
		<arg name="camera" value="$(arg camera)"/>
	</include>

	<!-- projection system -->
	<include file="$(find projector_interface)/launch/calibrate_and_circle.launch">
		<arg name="flip" value="false"/>
		<arg name="calibrate" value="false"/>
		<arg name="calibration_camera" value="$(arg camera)"/>
		<arg name="window_size" value="1"/>
	</include>

	<!-- glass -->

	<!-- 	fake face detection frame - assume a person is standing (approx 1.7 meters high)
	yaw (4th number) will need to be adjusted since glass' 0 is at magnetic north -->
	<node args="0 0 1.7 0 0 0 world /face_detection 100" name="face_detection" pkg="tf" type="static_transform_publisher"/>

	<include file="$(find projector_interface)/study/launch/glass.launch">
		<arg name="$(arg ip)"/>
		<arg name="$(arg port)"/>
	</include>
</launch>